roxie/udplib/udptrr.cpp

```diff
@@ -469,81 +409,80 @@ class CReceiveManager : implements IReceiveManager, public CInterface
+                            DBGLOG("UdpReceiver: received %s msg %u from node=%s", flowType::name(msg.cmd), msg.flowSeq, msg.sourceNode.getTraceText(ipStr).str());
                         }
                         UdpSenderEntry *sender = &parent.sendersTable[msg.sourceNode];
                         switch (msg.cmd)
                         {
                         case flowType::request_to_send:
-                            if (pendingRequests || currentRequester)
-                                enqueueRequest(sender);   // timeoutExpires does not change - there's still an active request. We have not given a new permission
-                            else
-                                newTimeout = okToSend(sender);
+                            enqueueRequest(sender);
                             break;
 
                         case flowType::send_completed:
-                            parent.inflight += msg.packets;
-                            if (noteDone(sender) && pendingRequests)   // This && looks wrong - noteDone returning false should mean we haven't seen the completed we wanted - so current timeout still applies. Or the one below is wrong...
-                                newTimeout = sendNextOk();
-                            else
-                                newTimeout = 5000;
+                            noteDone(sender);
                             break;
 
                         case flowType::request_to_send_more:
-                            parent.inflight += msg.packets;
-                            if (noteDone(sender))
-                            {
-                                if (pendingRequests)
-                                {
-                                    enqueueRequest(sender);
-                                    newTimeout = sendNextOk();
-                                }
-                                else
-                                    newTimeout = okToSend(sender);
-                            }
+                            noteDone(sender);
+                            enqueueRequest(sender);
                             break;
 
                         default:
                             DBGLOG("UdpReceiver: received unrecognized flow control message cmd=%i", msg.cmd);
                         }
-                        if (newTimeout)
-                            timeoutExpires = msTick() + newTimeout;
+                    }
+                    timeout = 5000;   // The default timeout is 5 seconds if nothing is waiting for response...
+                    if (pendingPermits)
+                    {
+                        unsigned now = msTick();
+                        for (UdpSenderEntry *finger = pendingPermits; finger != nullptr; )
+                        {
+                            if (now - finger->timeStamp >= 1000) //udpRequestToSendAckTimeout)
```

Comment:
Value of timeout didn't seem to make a lot of difference - or maybe higher was better because they are not really being lost! Needs more investigation